packages/hunspell.rb

```diff
@@ -3,32 +3,49 @@
 class Hunspell < Package
   description 'Hunspell is a spell checker and morphological analyzer library'
   homepage 'http://hunspell.github.io/'
-  version '1.6.2'
-  source_url 'https://github.com/hunspell/hunspell/archive/v1.6.2.tar.gz'
-  source_sha256 '3cd9ceb062fe5814f668e4f22b2fa6e3ba0b339b921739541ce180cac4d6f4c4'
+  version '1.7.0'
+  source_url 'https://github.com/hunspell/hunspell/archive/v1.7.0.tar.gz'
+  source_sha256 'bb27b86eb910a8285407cf3ca33b62643a02798cf2eef468c0a74f6c3ee6bc8a'
 
   binary_url ({
-    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/hunspell-1.6.2-chromeos-armv7l.tar.xz',
-     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/hunspell-1.6.2-chromeos-armv7l.tar.xz',
-       i686: 'https://dl.bintray.com/chromebrew/chromebrew/hunspell-1.6.2-chromeos-i686.tar.xz',
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/hunspell-1.6.2-chromeos-x86_64.tar.xz',
   })
   binary_sha256 ({
-    aarch64: '2122527ab9f428c6d88d22ea142c134bf648b933963178f368ce5e9832d93317',
-     armv7l: '2122527ab9f428c6d88d22ea142c134bf648b933963178f368ce5e9832d93317',
-       i686: '1e7bd3254580e4a211b1b8208e548675602290e9c7a1105ba66af6d4c9cee8da',
-     x86_64: '76e93eec7e0cec41707fd68f0f21351e94c25e8b993431f3bf400693d3de8251',
   })
 
   depends_on 'readline'
+  if ARGV[0] == 'install'
+    puts
+    puts "Enter your preferred language:"
+    puts "1 = English (US)"
+    puts "2 = French (FR)"
+    puts "3 = Spanish (ANY)"
+    puts "0 = Cancel"
+
+    while version = STDIN.gets.chomp
+      case version
+        when '1'
+          depends_on 'hunspell_en_us'
+          break
+        when '2'
+          depends_on 'hunspell_fr_fr'
+          break
+        when '3'
+          depends_on 'hunspell_es_any'
+          break
+        when '0'
+          abort
+      end
+    end
+  end
```

Comment:
It works like a charm, but as I install `hunspell` with en_US for instance and then install `hunspell_fr_fr`, the `depends_on 'hunspell'` in `hunspell_fr_fr` re-ask which language to install. Any idea how to avoid this like checking if `hunspell` is already installed (don't known how to do it in an other way than a `grep` in `crew list installed`)?

Reply:
I suppose you could remove the `depends_on 'hunspell'` from each language package and include the build there instead of here.  Or if you don't want to repeat for each language, create a hunspell_base.rb package and depend on that instead?

Reply:
I don't like both of them : in the first one, it will duplicate the build function and complexify maintenance, and in the second one it will multiply packages to maintain. But the second one seems better to me. Indeed : language packages won't change with hunspell version upgrade (to simplify, package url in header is here only to avoid chromebrew crashes). Having build duplicated inside these files will force to update them at each new unspell version, while depending on a new package will lead to update only 2 packages at each version upgrade. Hope I am clear...

Reply:
Not sure what to put where and how to design the dependency chain...

Reply:
Below is probably the most elegant way to solve this problem.
hunspell.rb
```
class Hunspell < Package
  description 'Hunspell is a spell checker and morphological analyzer library'
  homepage 'http://hunspell.github.io/'
  version '1.6.2'
  source_url 'https://github.com/hunspell/hunspell/archive/v1.6.2.tar.gz'
  source_sha256 '3cd9ceb062fe5814f668e4f22b2fa6e3ba0b339b921739541ce180cac4d6f4c4'
  version '1.7.0'
  source_url 'https://github.com/hunspell/hunspell/archive/v1.7.0.tar.gz'
  source_sha256 'bb27b86eb910a8285407cf3ca33b62643a02798cf2eef468c0a74f6c3ee6bc8a'

  if ARGV[0] == 'install'
    puts
    puts "Enter your preferred language:"
    puts "1 = English (US)"
    puts "2 = French (FR)"
    puts "3 = Spanish (ANY)"
    puts "0 = Cancel"

    while version = STDIN.gets.chomp
      case version
        when '1'
          depends_on 'hunspell_en_us'
          break
        when '2'
          depends_on 'hunspell_fr_fr'
          break
        when '3'
          depends_on 'hunspell_es_any'
          break
        when '0'
          abort
      end
    end
  end
end
```

Reply:
hunspell_base.rb:
```
class Hunspell_base < Package
  description 'Hunspell is a spell checker and morphological analyzer library'
  homepage 'http://hunspell.github.io/'
  version '1.7.0'
  source_url 'https://github.com/hunspell/hunspell/archive/v1.7.0.tar.gz'
  source_sha256 'bb27b86eb910a8285407cf3ca33b62643a02798cf2eef468c0a74f6c3ee6bc8a'

  binary_url ({
  })
  binary_sha256 ({
  })

  depends_on 'readline'

  def self.build
    system 'autoreconf -vfi'
    system './configure',
           "--prefix=#{CREW_PREFIX}",
           "--include=#{CREW_PREFIX}/include",
           "--libdir=#{CREW_LIB_PREFIX}",
           '--with-ui',
           '--with-readline'
    system "sed -i 's,/usr/share,#{CREW_PREFIX}/share,g' man/hunspell.1"
    system "sed -i 's,/usr/share,#{CREW_PREFIX}/share,g' src/tools/hunspell.cxx"
    system "sed -i 's,ncurses.h,#{CREW_PREFIX}/include/ncursesw/ncurses.h,' src/tools/hunspell.cxx"
    system 'make'
  end

  def self.install
    system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
  end
end
```
The instead of `depends_on 'hunspell'` in each language package, replace with `depends_on 'hunspell_base'`.
Hope that makes sense.

Reply:
Yes : that was exactly what I am designing and testing at the very moment. There will probably be a trap : it won't prevent you from installing only hunspell_base, but as it has no dictionary linked to it, its impact will be insignificant.

Reply:
Job's done. It works perfectly well here ! Thanks